src/model.py

```diff
@@ -147,9 +149,28 @@ def init_engine(self):
         self.setup_lora()
 
         # Create an AsyncLLMEngine from the config from JSON
-        self.llm_engine = AsyncLLMEngine.from_engine_args(
-            AsyncEngineArgs(**self.vllm_engine_config)
-        )
+        aync_engine_args = AsyncEngineArgs(**self.vllm_engine_config)
+        self.llm_engine = AsyncLLMEngine.from_engine_args(aync_engine_args)
+
+        # Create vLLM custom metrics
+        if not aync_engine_args.disable_log_stats:
+            try:
+                labels = {
+                    "model": self.args["model_name"],
+                    "version": self.args["model_version"],
+                }
+                self.metrics = VllmStatLogger(labels=labels)
+            except pb_utils.TritonModelException as e:
+                if "metrics not supported" in str(e):
+                    # Metrics are disabled at the server
+                    self.metrics = None
+                    self.logger.log_info("[vllm] Metrics not supported")
+                else:
+                    raise e
+
+            # Add vLLM custom metrics
+            if self.metrics:
+                self.llm_engine.add_logger("triton", self.metrics)
```

Comment:
Why not move into `try` block

Reply:
OK. I was thinking to leave less call in a try catch statement.

Reply:
Updated